core/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -155,6 +155,18 @@
         .put(SingleColumnTableType.ID, ImmutableSet.of()) // convertability handled in SingleColumnTableType
         .build();
 
+    /**
+     * Contains number conversions which are "safe" (= a conversion would not reduce the number of bytes
+     * used to store the value)
+     */
+    private static final ImmutableMap<Integer, Set<DataType>> SAFE_CONVERSIONS = ImmutableMap.<Integer, Set<DataType>>builder()
+        .put(BYTE.id(), ImmutableSet.of(SHORT, INTEGER, LONG, TIMESTAMP, FLOAT, DOUBLE))
+        .put(SHORT.id(), ImmutableSet.of(INTEGER, LONG, TIMESTAMP, FLOAT, DOUBLE))
+        .put(INTEGER.id(), ImmutableSet.of(LONG, TIMESTAMP, FLOAT, DOUBLE))
+        .put(LONG.id(), ImmutableSet.of(TIMESTAMP, DOUBLE))
```

Comment:
Can all longs be converted to double?

Reply:
Should be safe:

```
jshell> ((Long )Long.MAX_VALUE).doubleValue()
$2 ==> 9.223372036854776E18

jshell> ((Double)((Long )Long.MAX_VALUE).doubleValue()).longValue()
$5 ==> 9223372036854775807
```

Reply:
The conversion is close but not safe because we lose precision.

Reply:
I have mixed feelings about sacrificing precision over usability but it is probably what most people want.